hatbot.gs

```diff
@@ -0,0 +1,175 @@
+//requires OAuth1 library
+//initialize global variables
+var where_am_i = "London"
+
+//thresholds for temp and wind
+var too_hot_for_hat = 15;
+var too_windy_for_hat = 20;
+
+var hat_hot = false;
+var hat_windy = false;
+
+var bot_name = "🎩: ";
+
+/** Bank of tweet responses */
+var hot_and_windy = ["Get that hat out of here!","You'd have to be a FOOL to wear a hat.","NO HATS.", "I'd rather you didn't wear a hat.","This place ain't big enough for a hat."];
+var cold_and_windy = ["Better hold onto your hat, bud!","Try not to lose your hat!","Bit too blustery for a hat tbh.","I guess you could wear a wooly hat or something idk."];
+var hot_and_calm = ["It's too hot for a hat.","You'll look cool, but at what cost????????","A hat? In this economy? Absolutely not.", "It's not the time for a hat.","MAYBE a cap. Maybe."];
+var ideal_hat_scenario = ["Hats on, lads!","Wear a hat","ALL THE HATS!","Did somebody say HAT WEATHER?!","Finally, it's hat time.","No hats. JKS. Get that hat on b!", "Very excited to announce that it is time to wear a hat."];
+
+//test of emojis in strings
+function emoji_test(){
+  var text_out = bot_name + write_tweet();
+  Logger.log(text_out);
+}
+
+//twitter api keys
+var CONSUMER_KEY = '***';
+var CONSUMER_SECRET = '***';
+
+//twitter log in
+function getService() {
+  return OAuth1.createService('Twitter')
+      // Set the endpoint URLs.
+      .setAccessTokenUrl('https://api.twitter.com/oauth/access_token')
+      .setRequestTokenUrl('https://api.twitter.com/oauth/request_token')
+      .setAuthorizationUrl('https://api.twitter.com/oauth/authorize')
+
+      // Set the consumer key and secret.
+      .setConsumerKey(CONSUMER_KEY)
+      .setConsumerSecret(CONSUMER_SECRET)
+
+      // Set the name of the callback function in the script referenced
+      // above that should be invoked to complete the OAuth flow.
+      .setCallbackFunction('authCallback')
+  
+      // Using a cache will reduce the need to read from 
+      // the property store and may increase performance.
+      .setCache(CacheService.getUserCache())
+
+      // Set the property store where authorized tokens should be persisted.
+      .setPropertyStore(PropertiesService.getUserProperties());
+}
+
+function reset() {
+  var service = getService();
```

Comment:
Move this line to the top with variable assignment, then you don't need to keep calling it. (like on line 60)